File: JAVA/mooresvotingalgo.java
import java.util.Scanner;
class mooresvotingalgo{
    public static void main(String[] args) {
        int count=0;
        int major=-1;
        int freq=0;
        int x;

        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
            x=arr[i];
            if(freq==0){    
                major=x;
                freq=1;
            }
            else if(x==major){
                freq++;
            }
            else{
                freq--;
            }
        }
        count=0;
        for(int i=0;i<n;i++){
            if(major==arr[i]){
                count++;
            }
        }
        if(count>n/2){
            System.out.println(major);
        }
        else{
            System.out.println("No majority element");
        }
        sc.close();
    }
}

File: Python/mooresvotingalgo.py
n = int(input("Enter number of elements: "))
arr = []

print("Enter elements:")
for _ in range(n):
    num = int(input())
    arr.append(num)

candidate = arr[0]
count = 1

for i in range(1, n):
    if arr[i] == candidate:
        count += 1
    else:
        count -= 1
        if count == 0:
            candidate = arr[i]
            count = 1


count = 0
for i in range(n):
    if arr[i] == candidate:
        count += 1

if count > n // 2:
    print("Majority element is:", candidate)
else:
    print("No majority element found.")
